apis/syncset/v1alpha1/syncset_types.go

```diff
@@ -0,0 +1,50 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type SyncSetSpec struct {
+	GVKs []GVKEntry `json:"gvks,omitempty"`
+}
+
+type GVKEntry struct {
+	Group   string `json:"group,omitempty"`
+	Version string `json:"version,omitempty"`
+	Kind    string `json:"kind,omitempty"`
+}
+
+// +kubebuilder:resource:scope=Namespaced
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+
+// SyncSet is the Schema for the SyncSet API.
+type SyncSet struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   SyncSetSpec   `json:"spec,omitempty"`
+	Status SyncSetStatus `json:"status,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// SyncSetList contains a list of SyncSet.
+type SyncSetList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []SyncSet `json:"items"`
+}
+
+type SyncSetStatus struct {
+	ByPod []SyncSetStatusStatus `json:"byPod,omitempty"`
+}
+
+type SyncSetStatusStatus struct {
```

Comment:
this should be SyncSetPodStatus; may want to defer writing the status object until we have status reporting

Reply:
My intuition is that this by-pod status idea is good here, but I realize that `config` does not have that.

@maxsmythe wdyt?  It seems to me that building out byPod status for SyncSets is a good investment given cached data is part of enforcement, and we have bypod status for templates, constraints.

Reply:
Status reporting is a good idea generally. It can be added after-the-fact if needed without breaking backwards compatibility.

IMO adding it now signifies a commitment to add the feature in the near term. If you are okay with that then no reason to avoid frontloading the work.

If we add it, we should only include the most obvious fields (e.g. observedGeneration), until we are confident in our understanding of the caching problem.

Reply:
If we are ok with leaving it out in the beginning, I vote to omit it for now.

The only potential risk for the status that I see now is in making sure that the contract between the syncset_controller and the CacheManager provides the information necessary for the syncset_c to write the desired status to the syncset.

If observedGeneration is all we care about for the status, then there's nothing to worry about there.

Reply:
I can take the `status ` for now and we can all revisit it post alpha or as we near the controller work.